src/core/agent/agent.ts

```diff
@@ -240,11 +240,14 @@ class Agent {
         agentUrls.bootUrl
       );
       try {
-        await this.signifyClient.boot();
+        const bootResult = await this.signifyClient.boot();
+        if (bootResult.status !== 202 && bootResult.status !== 400) {
+          throw new Error(Agent.KERIA_NOT_BOOTED);
+        }
       } catch (e) {
         /* eslint-disable no-console */
         console.error(e);
-        throw new Error(Agent.KERIA_BOOT_FAILED);
+        throw e;
```

Comment:
the problem with this is fetch can throw a different error than the `Agent.KERIA_NOT_BOOTED` we are throwing. The UI needs this specific error.

Also, I forgot fetch has a `.ok` - lets use this for 202 in case a proxy changes it to a different 2xx code.

```
const bootResult = await this.signifyClient.boot().catch(e => {
  /* eslint-disable no-console */
  console.error(e);
  throw new Error(Agent.KERIA_BOOT_FAILED);
});

if (!bootResult.status.ok && bootResult.status !== 400) {
  /* eslint-disable no-console */
  console.warn(`Unexpected KERIA boot status returned: ${bootResult.status} ${bootResult.statusText}`);
  throw new Error(Agent.KERIA_NOT_BOOTED);
}

await this.signifyClient.connect().catch((e) => {
  /* eslint-disable no-console */
  console.error(e);
  throw new Error(Agent.KERIA_BOOTED_ALREADY_BUT_CANNOT_CONNECT);
});
```

^ I changed the style of the connect too to be consistent

Reply:
Oke. I just tested it in signify-ts. It works well with the logic above.